src/common/config.h

```diff
@@ -634,6 +638,133 @@ class Config {
+                                    size_t tsc_dequeue_start,
+                                    size_t tsc_dequeue_end) {
+    dequeue_stats_[dequeue_stats_id_].tsc_start_ = tsc_dequeue_start;
+    dequeue_stats_[dequeue_stats_id_].tsc_end_ = tsc_dequeue_end;
+    dequeue_stats_[dequeue_stats_id_].event_type_ = event_type;
+    dequeue_stats_id_++;
+  }
+
+  inline void UpdateDequeueTscWorker(int tid,
+                                     size_t frame_id,
+                                     size_t dequeue_tsc,
+                                     size_t valid_dequeue_tsc) {
+    total_worker_dequeue_tsc_[tid][frame_id] += dequeue_tsc;
+    total_worker_valid_dequeue_tsc_[tid][frame_id] += valid_dequeue_tsc;
+  }
+
+  inline void UpdateEnqueueTscWorker(int tid,
+                                     size_t frame_id,
+                                     size_t enqueue_tsc) {
+    total_worker_enqueue_tsc_[tid][frame_id] += enqueue_tsc;
+    worker_num_valid_enqueue_[tid][frame_id]++;
+  }
+
+  inline void LogEnqueueStatsWorker(int tid,
+                                    size_t symbol_id,
+                                    size_t start_tsc,
+                                    size_t end_tsc,
+                                    EventType event_type) {
+    size_t id = worker_enqueue_stats_id_[tid][symbol_id];
+    worker_enqueue_stats_[tid][symbol_id][id].tsc_start_ = start_tsc;
+    worker_enqueue_stats_[tid][symbol_id][id].tsc_end_ = end_tsc;
+    worker_enqueue_stats_[tid][symbol_id][id].event_type_ = event_type;
+    worker_enqueue_stats_id_[tid][symbol_id]++;
+  }
+
+  inline void LogDequeueStatsWorker(int tid,
+                                    size_t symbol_id,
+                                    size_t start_tsc,
+                                    size_t end_tsc,
+                                    EventType event_type) {
+    size_t id = worker_dequeue_stats_id_[tid][symbol_id];
+    worker_dequeue_stats_[tid][symbol_id][id].tsc_start_ = start_tsc;
+    worker_dequeue_stats_[tid][symbol_id][id].tsc_end_ = end_tsc;
+    worker_dequeue_stats_[tid][symbol_id][id].event_type_ = event_type;
+    worker_dequeue_stats_id_[tid][symbol_id]++;
+  }
+
+  // Task enqueue/dequeue start and end timestamps and task type
+  struct QueueTsStat {
+    EventType event_type_;
+    size_t tsc_start_ = 0;  // Unit = TSC cycles
+    size_t tsc_end_ = 0;    // Unit = TSC cycles
+  };
+
+  std::array<std::array<QueueTsStat, kMaxLoggingEventsMaster>, kMaxSymbols> enqueue_stats_;
+  std::array<QueueTsStat, kMaxLoggingEventsMaster> dequeue_stats_;
+  std::array<size_t, kMaxSymbols> enqueue_stats_id_ = {};
+  size_t dequeue_stats_id_ = 0;
+
+  std::array<std::array<size_t, kNumStatsFrames>, kMaxThreads>
+      total_worker_dequeue_tsc_ = {};
+  std::array<std::array<size_t, kNumStatsFrames>, kMaxThreads>
+      total_worker_enqueue_tsc_ = {};
+  std::array<std::array<size_t, kNumStatsFrames>, kMaxThreads>
```

Comment:
this is a lot of variables being added to config.  probably be best to make a class out of it

Reply:
requires lot of changes. we can discuss.